app/services/who_report_service.rb

```diff
@@ -0,0 +1,148 @@
+class WhoReportService
+  attr_reader :region, :period, :months, :repo, :dashboard_analytics
+  def initialize(region, period)
+    @region = region
+    @period = period
+    @months = period.downto(5).reverse
+    regions = region.facility_regions.to_a << region
+    @repo = Reports::Repository.new(regions, periods: period)
+    @dashboard_analytics = region.dashboard_analytics(period: period.type, prev_periods: 6)
+  end
+
+  def report
+    csv = CSV.generate(headers: true) { |csv|
+      csv << ["Facility Report #{period.to_date.strftime("%B %Y")}"]
+      csv << section_labels
+      csv << header_row
+      csv << district_row
+      facility_rows.each do |row|
+        csv << row
+      end
+    }
+  end
+
+  private
+
+  def section_labels
+    [
+      Array.new(12, nil),
+      "New Registrations",
+      Array.new(5, nil),
+      "Follow-up patients",
+      Array.new(5, nil),
+      "Treatment outcomes of patients under care",
+      Array.new(3, nil),
+      "Drug availability",
+      Array.new(2, nil)
+    ].flatten
+  end
+
+  def header_row
+    month_labels = months.map { |month| month.value.strftime("%b-%Y") }
+    [
+      "#",
+      "District",
+      "Facility",
+      "Facility type",
+      "Block",
+      "Active/Inactive (Inactive facilities have 0 BP measures taken)",
+      "Estimated hypertension opulation",
+      "Total registrations",
+      "Total assigned patients",
+      "Total lost to follow-up",
+      "Died",
+      "Total patients under care",
+      month_labels,
+      month_labels,
+      "BP controlled %",
+      "BP not controlled %",
+      "Missed visits %",
+      "Visit but no BP taken %",
```

Comment:
@kpethtel since these fields will now show patient numbers, let's rename these to:
* Patients with BP controlled
* Patients with BP not controlled
* Patients with a missed visit
* Patients with a visit but no BP taken

I've updated the Epic's description :)